fish-rust/src/common.rs

```diff
@@ -24,7 +24,7 @@ use num_traits::ToPrimitive;
 use once_cell::sync::Lazy;
```

Comment:
the `&mut &mut &mut` we may be able to get rid of by replacing (untested)

```
let zelf = scoped_push(zelf, ...);
```

with

```
let zelf = scoped_push(&*zelf, ...);
```

Reply:
I think there has to be a better way, along the lines of `std::remove_reference()` so that you don't have to do this manually at the call-site. Perhaps if we took `Borrow<Context>` instead of `Context` directly?